python/cuspatial/cuspatial/core/binpreds/feature_contains.py

```diff
@@ -87,47 +99,35 @@ def _compute_polygon_polygon_contains(self, lhs, rhs, preprocessor_result):
-        # Divide the linestring in half and test the point for containment
-        # in the polygon.
-        size_two = rhs.sizes == 2
-        if (size_two).any():
-            center_points = _linestrings_to_center_point(rhs[size_two])
-            size_two_results = _false_series(len(lhs))
-            size_two_results.iloc[rhs.index[size_two]] = (
-                _basic_contains_count(lhs, center_points) > 0
-            )
-            return size_two_results
-        else:
-            return _false_series(len(lhs))
-
     def _compute_polygon_linestring_contains(
         self, lhs, rhs, preprocessor_result
     ):
+        # Count the number of points in lhs that are properly contained by
+        # rhs
         contains = _basic_contains_count(lhs, rhs).reset_index(drop=True)
-        intersects = self._intersection_results_for_contains(lhs, rhs)
-
-        # If a linestring has intersection but not containment, we need to
-        # test if the linestring is in the interior of the polygon.
-        final_result = _false_series(len(lhs))
-        intersection_with_no_containment = (contains == 0) & (intersects != 0)
-        interior_tests = self._test_interior(
-            lhs[intersection_with_no_containment].reset_index(drop=True),
-            rhs[intersection_with_no_containment].reset_index(drop=True),
+
+        # Count the number of point intersections (line crossings) between
+        # lhs and rhs.
+        # Also count the number of perfectly overlapping linestring sections.
+        # Each linestring overlap counts as two point overlaps.
+        (
+            point_intersects_count,
+            linestring_intersects_count,
+        ) = self._intersection_results_for_contains_linestring(lhs, rhs)
+
+        # Subtract the length of the linestring intersections from the length
+        # of the rhs linestring, then test that the sum of contained points
+        # is equal to that adjusted rhs length.
+        rhs_sizes_less_line_intersection_size = (
+            rhs.sizes - linestring_intersects_count
         )
-        interior_tests.index = intersection_with_no_containment[
-            intersection_with_no_containment
-        ].index
-        # LineStrings that have intersection but no containment are set
-        # according to the `intersection_with_no_containment` mask.
-        final_result[intersection_with_no_containment] = interior_tests
-        # LineStrings that do not are contained if the sum of intersecting
-        # and containing points is greater than or equal to the number of
-        # points that make up the linestring.
-        final_result[~intersection_with_no_containment] = (
-            contains + intersects >= rhs.sizes
+        rhs_sizes_less_line_intersection_size[
+            rhs_sizes_less_line_intersection_size <= 0
+        ] = 1
+        final_result = contains + point_intersects_count == (
+            rhs_sizes_less_line_intersection_size
         )
+
         return final_result
```

Comment:
The reason we see this pattern so frequently is so that I can debug the `final_result` in the context of its precursors and inspect it while designing the predicate, which is very convenient. I'd rather not remove all of these from the codebase and then readd them whenever I need to see the inner workings of a particular result, ok?